paws_collector.js

```diff
@@ -447,6 +439,49 @@ class PawsCollector extends AlAwsCollector {
         });
     };
 
+    batchLogProcess(logs, privCollectorState, nextInvocationTimeout, asyncCallback) {
+        let collector = this;
+        return collector.processLog(logs, collector.pawsFormatLog.bind(collector), null, (err) => {
+           
+            if (err) {
+                // MS api return the logs but if build payload size > 10MB, our buildPayload method return Max payload issue. So we split logs array the in two part and call the processLog again.
+               
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
+                    let batchLogs = [];
+                    let halfLogs = [];
+                    let loglength = logs.length / 2;
+                    for (const index in logs) {
+                        console.log(index,logs[index]);
+                        if (index < loglength) {
+                            halfLogs.push(logs[index]);
+                            if (parseInt(index) === (loglength - 1)) {
+                                batchLogs.push(halfLogs);
+                            }
+                        }
+                        else {
+                            halfLogs = [];
+                            loglength = logs.length;
+                            halfLogs.push(logs[index]);
+                            if (parseInt(index) == logs.length-1) {
+                                batchLogs.push(halfLogs);
+                            }
+                        }
+                    }
+                    batchLogs.map(batchLog => {
+                         collector.batchLogProcess(batchLog, privCollectorState, nextInvocationTimeout, asyncCallback);
```

Comment:
we can't call `asyncCallback` here, becasue once the first batch is sent the callback will be returned and invocation will end without waiting for the 2nd batch to be sent. 
We need to send batches one by one and probably use `array.reduce()`

Reply:
Using async.reduce().